esiclient/tests/unit/v1/test_node_console.py

```diff
@@ -0,0 +1,70 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+from osc_lib import exceptions
+
+from esiclient.tests.unit import base
+from esiclient.v1 import node_console
+
+
+class TestNodeConsoleConnect(base.TestCommand):
+
+    def setUp(self):
+        super(TestNodeConsoleConnect, self).setUp()
+        self.cmd = node_console.NodeConsoleConnect(self.app, None)
+
+        self.node1 = {
+            "uuid": "11111111-2222-3333-4444-aaaaaaaaaaaa",
+            "name": "node1",
+            "console_enabled": False,
+            "console_info": None
+        }
+        self.node2 = {
+            "uuid": "11111111-2222-3333-4444-bbbbbbbbbbbb",
+            "name": "node2",
+            "console_enabled": True,
+            "console_info": {"type": "socat", "url": "tcp://10.10.10.10:1234"}
+            }
+        self.node3 = None
+
+    def test_take_action(self):
+
+        self.app.client_manager.baremetal.node.get_console.\
+            return_value = self.node1
+
+        arglist = ['node1']
+        verifylist = []
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        self.assertRaisesRegex(
+            exceptions.CommandError,
+            'ERROR: No console info for node1',
+            self.cmd.take_action, parsed_args)
+
+        self.app.client_manager.baremetal.node.get_console.\
+            return_value = self.node2
+
+        arglist = ['node2']
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        # result = self.cmd.take_action(parsed_args)
```

Comment:
I think this is the only node that `succeeds`; are you not running the test?

Reply:
We should get some message after it's connected to TCP. But it's returning none always. Looking for some alternative approach to get correct message connection refused or connected.
If you have any suggestion @tzumainn  please let me know. putting out some commented code.

Reply:
okay did something. It's working!